File: GSDRequirementsCSharp.Web/scripts/App/04-Models/Permission.js
var Models;
(function (Models) {
    var Permission = (function () {
        function Permission(data) {
            this.id = data['id'];
            this.profile = data['profile'];
            this.user = {
                'name': data['user']['contact']['name'],
                'email': data['user']['contact']['email'],
                'id': data['user']['id']
            };
        }
        return Permission;
    })();
    Models.Permission = Permission;
})(Models || (Models = {}));

File: GSDRequirementsCSharp.Domain/Models/Permission.cs
namespace GSDRequirementsCSharp.Domain
{
    using Domain;
    using Infrastructure.Persistence;
    using Models;
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    public class Permission : IEntity<Guid>
    {        
        public Guid Id { get; set; }

        public Guid ProjectId { get; set; }
        
        public Profile Profile { get; set; }

        public virtual Project Project { get; set; }
        
        public virtual User User { get; set; }

        public int UserId { get; set; }
    }
}
